File: Python/Leetcode/Array/h-Index.py
class Solution:
    def hIndex(seld, citations: list[int]) -> int:
        publications = len(citations)
        count = hIndex = 0
        
        while (publications):
            for i in range(len(citations)):
                
                if citations[i] >= publications:
                    count += 1
            
            if count >= publications:
                hIndex = max(hIndex, publications)
            
            count = 0
            publications -= 1
            
        return hIndex
    
    def hIndexII(self, citations: list[int]) -> int:
        citations.sort()
        h_index = len(citations)
        for i in citations:
            if i < h_index:
                h_index -= 1

        return h_index
    
if __name__ == '__main__':
    solution_instance = Solution()
    print(solution_instance.hIndex([3,0,6,1,5]))

File: C++/Leetcode/Array/hIndex.cpp
#include <iostream>
#include <vector>
#include <algorithm>

class Solution{
    public:
        int hIndex(std::vector<int>& citations){
            int hIndex = citations.size();
            std::sort(citations.begin(), citations.end());

            for (const auto& n : citations){
                if (n < hIndex){
                    hIndex--;
                }
            }
            return hIndex;
        }
    
    private:
        void printVector(std::vector<int>& nums){
            for (const auto& n : nums){
                std::cout << n << std::endl;
            }
        }
};

int main(){
    Solution solution_instance;
    std::vector<int> citations = {3,0,6,1,5}; 
    std::cout << solution_instance.hIndex(citations) << std::endl;
    return 0;
}